.github/workflows/qac.yml

```diff
@@ -0,0 +1,113 @@
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+
+  coverage:
+    name: Run Coverage 
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Cache .NET packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-dotnet-${{ hashFiles('**/*.csproj') }}
+          restore-keys: |
+            ${{ runner.os }}-dotnet-
+
+      - name: Restore dependencies
+        run: |
+          dotnet restore
+          dotnet tool restore
+
+      - name: Run coverage code 
+        run: |
+          dotnet tool install --global coverlet.console
+          dotnet build --no-incremental
+          dotnet test --results-directory ./coverage-report --collect:"XPlat Code Coverage"
+          dotnet reportgenerator -reports:"./coverage-report/*/coverage.cobertura.xml" -targetdir:"./coverage-report" -reporttypes:"MarkdownSummary"
+          # Coverlet report for sonarqube
+          coverlet tests/UnitTests/PlanthorWebApi.Application.Tests/bin/Debug/net8.0/PlanthorWebApi.Application.dll \
+            --target "dotnet" \
+            --targetargs "test --no-build" \
+            -f=opencover \
+            -o="./coverage-report/coverage.xml"
+              
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report
+          path: ./coverage-report/*
+      
+      - name: Code coverage sumarise 
+        run : |
+          echo "$(cat coverage-report/Summary.md)" >> $GITHUB_STEP_SUMMARY 
+
+  sonar-scan:
+    name: Run SonarScanner
+    runs-on: ubuntu-22.04
+    needs: [ coverage ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'zulu'
+
+      - name: Cache .NET packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-dotnet-${{ hashFiles('**/*.csproj') }}
+          restore-keys: |
+            ${{ runner.os }}-dotnet-
+
+      - name: Cache SonarScanner
+        uses: actions/cache@v4
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar-${{ hashFiles('./.github/workflows/qac.yml') }}
+
+      - name: Download coverage report
+        uses: actions/download-artifact@v4
+        with:
+          name: coverage-report
+          path: ./coverage-report
+
+      - name: Sonarqube Analytis
+        run: |
+          # mv -f  ./coverage-report/*/coverage.cobertura.xml ./coverage-report/coverage.cobertura.xml
```

Comment:
Do we need this mv command? Should we delete it? ?

Reply:
Will do